evm-contracts/src/v0.7/dev/OperatorForwarder.sol

```diff
@@ -4,23 +4,70 @@ pragma solidity ^0.7.0;
 import "../interfaces/OperatorInterface.sol";
 import "./ConfirmedOwner.sol";
 
-contract OperatorForwarder {
-  address public immutable authorizedSender1;
-  address public immutable authorizedSender2;
-  address public immutable authorizedSender3;
+contract OperatorForwarder is ConfirmedOwner{
+
+  mapping(address => bool) private s_authorizedSenders;
+  address[] private s_authorizedSenderList;
 
   address public immutable linkAddr;
 
+  event AuthorizedSendersChanged(
+    address[] senders
+  );
+
   constructor(
     address link
-  ) {
+  ) 
+    ConfirmedOwner(msg.sender)
+  {
     linkAddr = link;
-    authorizedSender1 = ConfirmedOwner(msg.sender).owner();
-    address[] memory authorizedSenders = OperatorInterface(msg.sender).getAuthorizedSenders();
-    authorizedSender2 = (authorizedSenders.length > 0) ? authorizedSenders[0] : address(0);
-    authorizedSender3 = (authorizedSenders.length > 1) ? authorizedSenders[1] : address(0);
   }
 
+  /**
+   * @notice Sets the fulfillment permission for a given node. Use `true` to allow, `false` to disallow.
+   * @param senders The addresses of the authorized Chainlink node
+   */
+  function setAuthorizedSenders(
+    address[] calldata senders
+  )
+    external
+    onlyOwner()
+  {
+    require(senders.length > 0, "Must have at least 1 authorized sender");
+    // Set previous authorized senders to false
+    uint256 authorizedSendersLength = s_authorizedSenderList.length;
+    for (uint256 i = 0; i < authorizedSendersLength; i++) {
```

Comment:
Is it faster to use e.g. uint32 here or it doesn't make any difference?

Reply:
This is a good question. I don't believe it will make a difference in memory here.

Reply:
Did some small tests on Rinkeby, `uint32` actually costs more than `uint256`.
- [`uint256` tx](https://rinkeby.etherscan.io/tx/0xc7c029eb73dfc071cf43f32bb26463dff61456009d74a1beccf4bdfeac695a12)
- [`uint32` tx](https://rinkeby.etherscan.io/tx/0x69f2351d808cbb9ab8b6ec5b038747cea854af223d2300eb920978aa4d75f863)

See [this StackExchange response](https://ethereum.stackexchange.com/a/3071/65131) as for why.

Reply:
Wow, that's mad. Thanks for the explanation